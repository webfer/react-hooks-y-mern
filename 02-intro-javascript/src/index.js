// file of work

const active = true;

// let msg = '';
//
// if (active) {
//   msg = 'active';
// } else {
//   msg = 'Unactive';
// }

//const msg = active ? 'Active' : 'unactive';
//const msg = active ? 'Active' : null;

//const msg = active && 'Active';
const msg = active && 'Active';
console.log(msg);
