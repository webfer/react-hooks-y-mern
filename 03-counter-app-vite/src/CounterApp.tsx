import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
  //console.log("Render");
  // API .... FETCH ....
  // Al hacer un cambio en el estado se vuelve a renderizar todo el FC = Functional component

  const [counter, setCounter] = useState(value);
  const handleAdd = () => {
    //   console.log(event);
    //setCounter((c) => c + 1); // Otra fora de actualizar el state, cuando no se conoce el valor de referencia (counter)
    setCounter(counter + 1);
  };

  const handleDecrease = () => setCounter(counter - 1);

  const handleRset = () => setCounter(value);

  return (
    <>
      <h1>CounterApp</h1>
      <h2> {counter} </h2>
      <button onClick={handleAdd}> +1</button>
      <button onClick={handleDecrease}> -1</button>
      <button aria-label="btn-reset" onClick={handleRset}>
        {' '}
        Reset
      </button>
    </>
  );
};

CounterApp.prototypes = {
  value: PropTypes.number.isRequired,
};
