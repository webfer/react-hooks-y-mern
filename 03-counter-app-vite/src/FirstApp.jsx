import React from 'react';
import PropTypes from 'prop-types';

// const getSaludo = () => {
//   let saludo = {
//     name: "Fernando",
//     lastName: "Castro",
//   };
//   return saludo;
// };

export const FirstApp = ({
  title,
  name,
  subTitle,
  // subTitle = "Sub-titulo que pasa por defecto",
}) => {
  return (
    <>
      {/* <h1>Hello, {getSaludo().name + " " + getSaludo().lastName}</h1> */}
      <h1 data-testid="test-title">{title}</h1>
      <h2>Hello, {name}</h2>
      <h3>{subTitle}</h3>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  name: 'No hay nombre',
  subTitle: 'No hay subtitle',
  // title: "Este es el titulo",
};
