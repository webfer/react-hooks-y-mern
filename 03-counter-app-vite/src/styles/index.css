html,
body {
  background-color: #21232a;
  color: #fff;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 1.3rem;
  padding: 70px;
}

button {
  font-size: 1.3rem;
  margin-right: 5px;
  padding: 15px;
}
