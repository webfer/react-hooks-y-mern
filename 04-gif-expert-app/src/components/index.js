export * from './AddCategory';
export * from './GifExpertApp';
export * from './GifGrid';
export * from './GifItem';
