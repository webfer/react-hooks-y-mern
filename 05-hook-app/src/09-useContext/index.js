export * from './HomePage';
export * from './AboutPage';
export * from './LoginPage';
export * from './Navbar';
