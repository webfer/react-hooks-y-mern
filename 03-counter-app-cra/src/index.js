import React from 'react';
import { createRoot } from 'react-dom/client';
// import PrimeraApp from './components/PrimeraApp';
import CounterApp from './components/CounterApp';

import './index.css';

const divRoot = document.getElementById('app');
const root = createRoot(divRoot);

root.render(<CounterApp value={10} />);
