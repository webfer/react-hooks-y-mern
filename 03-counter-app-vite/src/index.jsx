import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';

import './styles/index.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <FirstApp name="Fernando" title="Hola, soy Vegeta!" /> */}
    <CounterApp value={10} />
  </React.StrictMode>
);
