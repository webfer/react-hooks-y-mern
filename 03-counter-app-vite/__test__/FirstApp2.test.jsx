import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Testing on <FirstApp/>', () => {
  const title = 'Hola soy Guku';
  const subTitle = 'Soy un sub-titulo';
  test('Debe de hacer match cion el snapshot', () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test('Debe de mostrar el mensaje "Hola, soy Guku"', () => {
    render(<FirstApp title={title} />);
    // screen.debug();
    // expect(screen.getByText(title)).not.toBeTruthy(); // With negative statement
    expect(screen.getByText(title)).toBeTruthy();
  });

  test('Debe de mostrar el titulo en un H1', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test('Debe mostrar el subtitulo enviado por PROPS', () => {
    render(<FirstApp title={title} subTitle={subTitle} />);
    expect(screen.getAllByText(subTitle).length).toBe(1);
  });
});
