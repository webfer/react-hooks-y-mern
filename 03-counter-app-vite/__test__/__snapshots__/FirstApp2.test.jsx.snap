// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on <FirstApp/> Debe de hacer match cion el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy Guku
  </h1>
  <h2>
    Hello, 
    No hay nombre
  </h2>
  <h3>
    No hay subtitle
  </h3>
</div>
`;
